src/dspbase.jl

```diff
@@ -118,75 +118,319 @@ function deconv(b::StridedVector{T}, a::StridedVector{T}) where T
 end
 
 # padded must start at index 1, but u can have arbitrary offset
-function _zeropad!(padded::AbstractVector, u::AbstractVector)
-    datasize = length(u)
+@inline function _zeropad!(
+    padded::AbstractVector,
+    u::AbstractVector,
+    data_dest = (1,),
```

Comment:
I put a `Tuple` type bound on it, to emphasize that a `tuple` is expected. It actually needs to be a tuple not just because it comes from size, but mostly to work with `CartesianIndices`. The reason the `Vector` method also needs a `Tuple` is to have the same interface as the ND method.